application/helpers/common_helper.php

```diff
@@ -3999,6 +3999,26 @@ function questionAttributes($returnByName=false)
         "help"=>gT('Use button group or radio list'),
         "caption"=>gT('Display type'));
 
+        /**
+         * New event to allow plugin to add own question attribute (settings)
+         * Using $event->append('questionAttributes', $questionAttributes);
+         * $questionAttributes=[
```

Comment:
Actual solution : all existing attributes are defined by an array

````
        $qattributes["display_type"]=array(
        "types"=>"YG",
        'category'=>gT('Display'),
        'sortorder'=>90,
        'inputtype'=>'buttongroup',
        'options'=>array(0=>gT('Button group','unescaped'),
        1=>gT('Radio list','unescaped')),
        'default'=>0,
        "help"=>gT('Use button group or radio list'),
        "caption"=>gT('Display type'));
````

Then i just add a way to allow plugin to add a new attributes, using same system.

See https://gist.github.com/Shnoulle/f959ffba582a304bf13c110094dad996